tests/Unit/Domain.LinnApps.Tests/PurchaseOrderDeliveryServiceTests/WhenUpdatingDetailsRemovingFromDeliveryTotal.cs

```diff
@@ -0,0 +1,138 @@
+                    AuthorisedAction.PurchaseOrderUpdate, Arg.Any<IEnumerable<string>>())
+                .Returns(true);
+
+            this.MiniOrderRepository.FindById(this.order.OrderNumber).Returns(new MiniOrder { });
+            this.result = this.Sut.UpdateDeliveriesForOrderLine(
+                this.order.OrderNumber,
+                this.line.Line,
+                this.updatedDeliveries,
+                new List<string>());
+        }
+
+        [Test]
+        public void ShouldReturnResult()
+        {
+            //Delivery fields
+            this.result.Count().Should().Be(1);
+            var updateData = this.updatedDeliveries.First();
+            var updated = this.result.First();
+            updated.DeliverySeq.Should().Be(1);
+            updated.OurDeliveryQty.Should().Be(updateData.OurDeliveryQty);
+            updated.OrderDeliveryQty.Should().Be(updateData.OurDeliveryQty / this.line.OrderConversionFactor);
+            updated.OurUnitPriceCurrency.Should().Be(this.line.OurUnitPriceCurrency);
+            updated.DateRequested.Should().Be(updateData.DateRequested);
+            updated.DateAdvised.Should().Be(updateData.DateAdvised);
+            updated.CallOffDate.Should().BeCloseTo(DateTime.Now, new TimeSpan(0, 1, 0));
+            updated.NetTotalCurrency.Should().Be(
+                Math.Round(
+                    updateData.OurDeliveryQty.GetValueOrDefault() * this.line.OurUnitPriceCurrency.GetValueOrDefault(),
+                    2));
+            updated.VatTotalCurrency.Should().Be(this.vatAmount);
+            updated.DeliveryTotalCurrency.Should().Be(
+                Math.Round(
+                    this.line.OrderUnitPriceCurrency.GetValueOrDefault() * updated.OurDeliveryQty.GetValueOrDefault(),
+                    2) + this.vatAmount);
+            updated.BaseOurUnitPrice.Should().Be(updateData.BaseOurUnitPrice);
+            updated.BaseOrderUnitPrice.Should().Be(updateData.BaseOrderUnitPrice);
+            updated.BaseNetTotal.Should().Be(Math.Round(
+                updateData.OurDeliveryQty.GetValueOrDefault() * this.line.BaseOurUnitPrice.GetValueOrDefault(),
+                2));
+            updated.BaseVatTotal.Should().Be(this.baseVatAmount);
+            updated.BaseDeliveryTotal.Should().Be(
+                this.baseVatAmount + (updateData.OurDeliveryQty.GetValueOrDefault()
+                                      * this.line.BaseOurUnitPrice.GetValueOrDefault()));
+            updated.QuantityOutstanding.Should().Be(
+              10);
+            updated.RescheduleReason.Should().Be("REQUESTED");
+            updated.AvailableAtSupplier.Should().Be(updateData.AvailableAtSupplier);
+
+            //Sum total of the delivery quantities
+            line.OrderQty.Should().Be(10);
+        }
+
+        [Test]
+        public void ShouldUpdateMiniOrder()
+        {
+            this.MiniOrderRepository.Received().FindById(this.order.OrderNumber);
```

Comment:
one last thing - the UpdateMiniOrder() method in the PurchaseOrderDeliveryService doesn't actually update the qty on the mini order, so as it stands this will fall out of sync with the  PL_ORDER unless you change it. Are we accepting that mini_order is just to die now? If not you should make sure you update the mini order qty in step with the pl_order_detail qties too

Reply:
i think we need to keep it in sync for a bit longer

Reply:
I think that's a question for Richard but I'll keep it in sync for now